### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by gjuarez14**
Part 2 finished

### CODE REVIEW

File: src/classes/PostTilesService.cls [hunk 1]
```diff
@@ -0,0 +1,92 @@
+public class PostTilesService {
+
+    private Integer EXCERPT_LENGTH = 140;
+
+    public PostTilesService(){
+
+    }
+
+    public List<Post__c> allPosts(){
+        List<Post__c> allPosts = Database.query(
```
Comment (rjulianialtimetrik) [inline src/classes/PostTilesService.cls]: Please make sure the user has the proper FLS and CRUD permissions before querying. You might want to google for "Check CRUD and FLS in Apex"

File: src/classes/PostTilesService.cls [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        if(Schema.sObjectType.Contact.fields.Name.isAccessible()){
+            fields.add('Writer__r.name');
+        }
+        if(Schema.sObjectType.Attachment.fields.Id.isAccessible()){
+            fields.add('(Select id from Attachments limit 1)');
+        }
+        return fields;
+    }
+    
+    private List<String> accessibleFields(String sObjectType, List<String> desiredFields){
```
Comment (gjuarez14) [inline src/classes/PostTilesService.cls]: We created a method that filters all the accessible fields before doing the query.

File: src/classes/PostTilesService.cls [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    
+    public List<Post__c> filterPostsByWriterAndTag(Id writerId, String tag){
+        List<Post__c> postsWithWriterAndTag = Database.query(
+            'SELECT Id, Name, Content__c, Tags__c, Writer__r.name, Status__c, (Select id from Attachments limit 1) '+
+            'FROM Post__c WHERE (Writer__r.Id = :writerId AND Tags__c LIKE \'%'+tag+'%\') LIMIT 10000'
+        );
+        return postsWithWriterAndTag;
+    }
+    
+    public List<Tile> tilesFromPosts(List<Post__c> posts){
```
Comment (gjuarez14) [inline src/classes/PostTilesService.cls]: We also checked for accessibility before adding the attributes to the Tile

File: src/classes/PostTilesService.cls [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    public PostTilesService(){
+        
+    }
+    
+    public List<Post__c> allPosts(){
+        List<Post__c> allPosts = new List<Post__c>();
+        String fields = String.join(accessibleTileFields(), ',');
+        if(Schema.sObjectType.Post__c.fields.Name.isAccessible()){
+            allPosts = Database.query(
+                'SELECT '+ fields +' FROM Post__c LIMIT 10000'
```
Comment (nataliaortiz2017) [inline src/classes/PostTilesService.cls]: We added limit to all the queries

File: src/pages/Post_List_Tiles.page [hunk 5]
```diff
@@ -0,0 +1,133 @@
+<apex:page sidebar="false" controller="PostTilesController" docType="html-5.0" readOnly="true">
```
Comment (nataliaortiz2017) [inline src/pages/Post_List_Tiles.page]: We decided to make the page read only since this allows to retrieve more records.

**Pull request merged**